apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -1662,6 +1675,247 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_inline_paths_with_subselections() {
```

Comment:
worth testing this with variables? i've had use cases something like 
```
body: """
id: $this.id
$args.input {
  title
  body
}
"""

Reply:
This was interesting! https://github.com/apollographql/router/pull/6076/commits/dede80cb4ee87ca67f2cfa7250cbbc4078d5cd7c?diff=unified&w=0

Reply:
As an example of something I learned while writing these tests, you can flatten more than one level of nesting using this syntax:
```rust
            assert_eq!(
                selection!(
                    r#"
                    id: $this.id
                    $args { .input { title body } extra }
                    from
                "#
                )
                .apply_with_vars(&data, &vars),
                (
                    Some(json!({
                        "id": 1234,
                        "title": "The capital of Australia",
                        "body": "Canberra",
                        "extra": "extra",
                        "from": "data",
                    })),
                    vec![],
                ),
            );
```